src/profiling/with_perf.md

```diff
@@ -0,0 +1,294 @@
+# Profiling with perf
+
+sThis is a guide for how to profile rustc with perf.
```

Comment:
Perhaps add a link to one of the following?
- https://perf.wiki.kernel.org/index.php/Main_Page
- http://www.brendangregg.com/perf.html